plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/outline/MetaTypeAwareComparator.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ui.outline;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.n4js.ts.types.TypesPackage;
+import org.eclipse.xtext.ui.editor.outline.IOutlineNode;
+import org.eclipse.xtext.ui.editor.outline.actions.SortOutlineContribution;
+import org.eclipse.xtext.ui.editor.outline.impl.EObjectNode;
+import org.eclipse.xtext.ui.editor.outline.impl.OutlineFilterAndSorter.IComparator;
+import org.eclipse.xtext.ui.editor.preferences.IPreferenceStoreAccess;
+
+import com.google.inject.Inject;
+
+/**
+ *
+ */
+public class MetaTypeAwareComparator implements IComparator {
```

Comment:
I know. The problem is: I'm not quite sure how to add several sorters in the outline view. The API is a little bit strange. In the long run, we can change that. For the time being I wanted the things to be sorted as I did it. It is a pragmatic solution.

Oh, and the order is far from random! It is simply metatype based now, i.e. fields are sorted before methods.